### ISSUE TITLE
arch-chroot_install.sh script needs improvements

### ISSUE THREAD

**Issue by mrwerdo** [NUbots/NUbots#794]
1. `ROBOT_NUMBER` is used before being checked.
   https://github.com/NUbots/NUbots/blob/29431e6595c4e61a8d1acc8d6a8b7e9ebb79463c/doc/ArchInstall/arch-chroot_install.sh#L8
1. Arch install script is searching for the incorrect interface name (wlan0 instead of wlp58s0) on one of the robots. https://github.com/NUbots/NUbots/blob/29431e6595c4e61a8d1acc8d6a8b7e9ebb79463c/doc/ArchInstall/arch-chroot_install.sh#L11

The two problems caused the Wifi and Ethernet connection to break on the robot. The `ROBOT_NUMBER` needs to be checked in the script in case it isn't provided. It's an error if it's not provided because it's used in the `Address=` in the network config files. If. the robot doesn't have a number, then it doesn't have an IP address.

Searching for the interface name was failing for some reason. The variable was assigned an empty string and when substituted into the network config files was also causing the wifi interface to stop working.

**Comment by ysims** [comment]
@ZeoNyph @sebastianbrooker2 not sure if you want this closed or not yet, but with the phasing out of the NUgus it's not relevant anymore.

**Comment by ZeoNyph** [comment]
Yeah we will have to close this since we're no longer setting up the compute boards from scratch with the K1s. Any relevant configuration done here will be done in SystemConfiguration
